### PR TITLE
Enable https tests

### PR DESCRIPTION
**Pull request by fao89**
closes #8677

### CODE REVIEW

File: templates/github/.github/workflows/scripts/install.sh.j2 [hunk 1]
```diff
@@ -146,6 +146,13 @@ fi
 ansible-playbook build_container.yaml
 ansible-playbook start_container.yaml
 
+# Hack: adding pulp CA to certifi.where()
+sudo docker cp pulp:/etc/pulp/certs/ca.crt /usr/local/share/ca-certificates/pulp_ca.crt
+CERT=$(python -c 'import certifi; print(certifi.where())')
```
Comment (fao89) [inline templates/github/.github/workflows/scripts/install.sh.j2:151]: This is Mozilla CA bundle, and I'm appending pulp CA to it

File: templates/github/.github/workflows/scripts/script.sh.j2 [hunk 2]
```diff
@@ -94,7 +94,7 @@ if [[ "$TEST" = 'bindings' || "$TEST" = 'publish' ]]; then
   gem install --both ./{{ plugin_snake }}_client-0.gem
   cd ..
 {%- endif %}
-  ruby $REPO_ROOT/.ci/assets/bindings/test_bindings.rb
+  SSL_CERT_FILE=/usr/local/share/ca-certificates/pulp_ca.crt ruby $REPO_ROOT/.ci/assets/bindings/test_bindings.rb
```
Comment (fao89) [inline templates/github/.github/workflows/scripts/script.sh.j2:97]: Ruby doesn't use certifi lib, so I had to use an OpenSSL env var to point to the CA

File: templates/github/.github/workflows/scripts/install.sh.j2 [hunk 3]
```diff
@@ -146,6 +146,24 @@ fi
 ansible-playbook build_container.yaml
 ansible-playbook start_container.yaml
 
+echo ::group::SSL
+# Copy pulp CA
+sudo docker cp pulp:/etc/pulp/certs/pulp_webserver.crt /usr/local/share/ca-certificates/pulp_webserver.crt
+sudo docker cp pulp:/etc/pulp/certs/pulp_webserver.crt /etc/ssl/certs/pulp_webserver.crt
+sudo docker cp pulp:/etc/pulp/certs/pulp_webserver.key /etc/ssl/private/pulp_webserver.key
```
Comment (mdellweg) [inline templates/github/.github/workflows/scripts/install.sh.j2:153]: I think, we don't need the key here. We should not expose it more than needed.

**Review by mdellweg — APPROVED** [review]
Quite a change... But it looks good to me!

**Comment by fao89** [comment]
Since this is approved should we merge it?
About the single container part https://github.com/pulp/pulp-oci-images/pull/73 I really don't know what to do, maybe we should decide on the next CI/CD meeting

**Comment by mdellweg** [comment]
One last thing to consider: If we apply this template to old release branches, will they have a problem?
There is nothing that sticks out to me, however.

**Comment by daviddavis** [comment]
We could open a pr and find out?

**Comment by fao89** [comment]
> One last thing to consider: If we apply this template to old release branches, will they have a problem?
> There is nothing that sticks out to me, however.

since we are deciding to have HTTP and HTTPS tags, I can change this PR for https be optional

**Comment by daviddavis** [comment]
I think that makes sense. Some plugins (like pulp_ansible) required updates to their tests since they do pulp-to-pulp syncs, etc. and these test changes might be hard to backport to older release branches.

I would recommend we default to https being true.

**Comment by fao89** [comment]
Blocked by https://github.com/pulp/pulp-oci-images/pull/73 or https://github.com/pulp/pulp-oci-images/pull/84

**Comment by daviddavis** [comment]
Hey, can you attach this to https://github.com/pulp/plugin_template/issues/403?

File: plugin-template [hunk 4]
```diff
@@ -33,6 +33,7 @@ DEFAULT_SETTINGS = {
     'docker_fixtures': False,
     'docs_test': True,
     'issue_tracker': 'redmine',
+    'pulp_scheme': 'http',
```
Comment (fao89) [inline plugin-template:36]: made `http` default, for now, so I can merge it.
After merging https://github.com/pulp/pulp-oci-images/pull/73 or https://github.com/pulp/pulp-oci-images/pull/84
we can change it to `https`

**Pull request merged**